app/src/auto_validator/core/utils/utils.py

```diff
@@ -4,31 +4,52 @@
 import os
 import subprocess
 
-import bittensor as bt
-import paramiko
+import bittensor as bt  # type: ignore
 import requests
 import yaml
-from django.conf import settings
-from django.shortcuts import redirect, render
-from scp import SCPClient
+from django.conf import settings  # type: ignore
+from django.http.response import HttpResponse, HttpResponseRedirect  # type: ignore
+from django.shortcuts import redirect, render  # type: ignore
+from git import GitCommandError, Repo
 
-from ..models import Hotkey, Subnet
+from ..models import Subnet
+from .ssh import SSH_Manager
 
-GITHUB_URL = settings.SUBNETS_INFO_GITHUB_URL
+GITHUB_SUBNETS_CONFIG_PATH = settings.SUBNETS_INFO_GITHUB_URL
 LOCAL_SUBNETS_SCRIPTS_PATH = settings.LOCAL_SUBNETS_SCRIPTS_PATH
+GITHUB_SUBNETS_SCRIPTS_PATH = settings.GITHUB_SUBNETS_SCRIPTS_PATH
+BITTENSOR_WALLET_PATH = settings.BITTENSOR_WALLET_PATH
+BITTENSOR_WALLET_NAME = settings.BITTENSOR_WALLET_NAME
+BITTENSOR_HOTKEY_NAME = settings.BITTENSOR_HOTKEY_NAME
+VALIDATOR_SECRET_VALUE_TYPES = settings.VALIDATOR_SECRET_VALUE_TYPES
 
 
-def fetch_and_compare_subnets(request):
-    response = requests.get(GITHUB_URL, timeout=30)
+def fetch_and_compare_subnets(request: requests.Request) -> requests.Response | HttpResponse | HttpResponseRedirect:
+    # Clone the subnet scripts repository using GitPython
+    try:
+        if os.path.exists(LOCAL_SUBNETS_SCRIPTS_PATH):
+            # If the directory already exists, remove it
+            subprocess.run(["rm", "-rf", LOCAL_SUBNETS_SCRIPTS_PATH], check=True)
```

Comment:
there is a python native way of deleting files, shutil or something. Use that, or os.unlink, but not bash